smtlib-backends-process/src/SMTLIB/Backends/Process.hs

```diff
@@ -31,22 +30,10 @@ import qualified Data.ByteString.Lazy.Char8 as LBS
 import Data.Default (Default, def)
 import GHC.IO.Exception (IOException (ioe_description))
 import SMTLIB.Backends (Backend (..))
-import System.Exit (ExitCode (ExitFailure))
 import qualified System.IO as IO
-import System.Process.Typed
-  ( Process,
-    getStderr,
-    getStdin,
-    getStdout,
-    mkPipeStreamSpec,
-    setStderr,
-    setStdin,
-    setStdout,
-    startProcess,
-    stopProcess,
-    waitExitCode,
-  )
-import qualified System.Process.Typed as P (proc)
+-- we use process instead of typed-process because of a race condition in
+-- killing a process: https://github.com/fpco/typed-process/issues/38
```

Comment:
```suggestion
```
I'd leave this matter for the change log alone. It would be fine to mention the issue there.

This is not the only reason though. Design-wise it makes sense to go with the leaner solution that works. `typed-process` has a non-trivial implementation, and pulls some extra dependencies, which adds up to the maintenance effort.